src/thing_creature.c

```diff
@@ -5409,48 +5409,70 @@ void controlled_creature_pick_thing_up(struct Thing *creatng, struct Thing *pick
 
 void controlled_creature_drop_thing(struct Thing *creatng, struct Thing *droptng)
 {
-    creature_drop_dragged_object(creatng, droptng);
+    if (droptng->class_id == TCls_Creature)
+    {
+        stop_creature_being_dragged_by(droptng, creatng);
+    }
+    else
+    {
+        creature_drop_dragged_object(creatng, droptng);
+    }
     clear_messages_from_player(-81);
     clear_messages_from_player(-86);
     unsigned short smpl_idx, pitch;
-    switch(droptng->class_id)
+    struct SlabMap *slb = get_slabmap_for_subtile(droptng->mappos.x.stl.num, droptng->mappos.y.stl.num);
+    if (slb->kind == SlbT_WATER)
     {
-        case TCls_Object:
+        smpl_idx = (rand() % (24 - 22)) + 21;
+        pitch = 75;
+    }
+    else
+    {
+        switch(droptng->class_id)
         {
-            smpl_idx = 992;
-            struct ObjectConfigStats* objst = get_object_model_stats(droptng->model);
-            switch (objst->genre)
+            case TCls_Object:
             {
-                case OCtg_WrkshpBox:
-                case OCtg_SpecialBox:
-                {
-                    pitch = 25;
-                    break;
-                }
-                case OCtg_Spellbook:
-                {
-                    pitch = 90;
-                    break;
-                }
-                default:
+                smpl_idx = 992;
+                struct ObjectConfigStats* objst = get_object_model_stats(droptng->model);
+                switch (objst->genre)
                 {
-                    pitch = 0;
-                    break;
+                    case OCtg_WrkshpBox:
+                    case OCtg_SpecialBox:
+                    {
+                        pitch = 25;
+                        break;
+                    }
+                    case OCtg_Spellbook:
+                    {
+                        pitch = 90;
+                        break;
+                    }
+                    default:
+                    {
+                        pitch = 0;
+                        break;
+                    }
                 }
+                break;
+            }
+            case TCls_Creature:
+            {
+                smpl_idx = (rand() % (20 - 18)) + 17;
```

Comment:
could those 2 rand % lines be replaced with a more descriptive macro?
kinda hard to read as is

Reply:
I believe there's functions for this to use.